### ISSUE TITLE
Invalid assertions in `SessionPlayerConnector`

### ISSUE THREAD

**Issue by georgi-neykov-hub** [google/ExoPlayer#8047]
### [REQUIRED] Issue description
I am using the new media2 extension for Exoplayer (v2.12.0) and I am trying to use the new playlist APIs in a `Player` connected via a `SessionPlayerConnector` to a `MediaLibrarySession`. The `SessionPlayer` APIs from AndroidX prohibit calling `SessionPlayer.setMediaItem` with a `null` `MediaItem` argument and calling `SessionPlayer.setPlaylist()` with a `null` or empty `List<MediaItem>`, but is perfectly fine to call `SessionPlayer.removeMediaItem()` until the playlist is empty.

In my use case I want to update the playlist of a given `ExoPlayer` instance with a diff change script (add/remove/change/move) applied via the existing `Player` playlist modification APIs. 
Everything is mostly fine (#8046 ) until the playlist goes empty and an assertion in `SessionPlayerConnector.handlePlaylistChangedOnHandler()` kicks in:

```java
...
if (notifyCurrentMediaItem) {
            Assertions.checkNotNull(
                currentMediaItem, "PlaylistManager#currentMediaItem() cannot be changed to null");

            callback.onCurrentMediaItemChanged(SessionPlayerConnector.this, currentMediaItem);
...
```

The code above results in inability to have the player playlist cleared either directly via the `com.google.android.exoplayer2.Player` or the APIs from `androidx.media2.common.SessionPlayer`  and have the updates to state reported to media2 related classes. 

Having this issue together with (#8011) greatly makes the extension unusable if someone wants to use both the playlist APIs and media2 extension introduced in 2.12.0.

### [REQUIRED] Reproduction steps
1. Instantiate and configure a MediaSession, link it to a `SessionPlayerConnector` with a `Player` instance
2. Add playlist items to the `Player`, either directly or via the `SessionPlayer` API
3. Clear the playlist directly through the `Player`, or by calling `SessionPlayer.removeMediaItem(0)` until the playlist is empty.

### [REQUIRED] Link to test content
The issue is not related to media file decoding or playback.

### [REQUIRED] A full bug report captured from the device
[assertion stacktrace.txt](https://github.com/google/ExoPlayer/files/5331996/assertion.stacktrace.txt)

### [REQUIRED] Version of ExoPlayer being used
2.12.0

### [REQUIRED] Device(s) and version(s) of Android being used
Any Android-powered device or emulator complying with the minSdk of the Exoplayer library.

**Comment by tonihei** [comment]
@sim0629  Could you take a look?

**Comment by sim0629** [comment]
I'll be on leave from tomorrow. Sungsoo, could you take a look?

**Comment by ojw28** [comment]
@SungsooLim - We should try and get this fixed for a minor release. It's a bit unclear what `SessionPlayer.PlayerCallback. onCurrentMediaItemChanged` should be called with when the playlist becomes empty. Should it be sent an empty `MediaItem`?

**Comment by SungsooLim** [comment]
When MediaItem or Playlist becomes becomes null or empty, then the state of SessionPlayer should become idle.
Let me investigate and think a little more for the current status and the proper fix.

**Comment by SungsooLim** [comment]
Fixed.

**Comment by ojw28** [comment]
Fixed in https://github.com/google/ExoPlayer/commit/2f1c0a5d0d556f9896af43b61ff0d8f04f31de30. This will be included in the `2.12.2` release, which is due in the next couple of weeks.
